kyuubi-main/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala

```diff
@@ -65,6 +66,14 @@ class KyuubiSessionImpl(
   private var client: TCLIService.Client = _
   private var remoteSessionHandle: TSessionHandle = _
 
+  private val isClusterMode = sessionConf.getOption(SPARK_SUBMIT_DEPLOY_MODE)
+    .map(_.toLowerCase(Locale.ROOT)).contains("cluster")
+
+  if (isClusterMode) {
+    sessionConf.set(SPARK_YARN_MAX_APP_ATTEMPTS, "1")
```

Comment:
Why should it be 1?

Reply:
Prevent AM restarts when the engine reaches its maximum idle time and terminated.

Reply:
AM will restart when it terminated w/ FINISHED state?

Reply:
enen

Reply:
What makes it different from other regular spark applications? I guess they are the same

Reply:
I made a mistake,  AM didn't w/ FINISHED state when it terminated. I will fix it.